lxd/migrate_storage_volumes.go

```diff
@@ -349,7 +349,7 @@ func (c *migrationSink) DoStorage(state *state.State, projectName string, poolNa
 		for _, sourceSnap := range sourceSnapshots {
 			sourceSnapshotComparable = append(sourceSnapshotComparable, storagePools.ComparableSnapshot{
 				Name:         sourceSnap.GetName(),
-				CreationDate: time.Unix(sourceSnap.GetCreationDate(), 0),
+				CreationDate: time.Unix(0, sourceSnap.GetCreationDate()),
```

Comment:
This looks wrong, the creation date is now passed to the nanseconds argument of time.Unix().

Reply:
Oh I see you're encoding the nanoseconds from 1970 into the CreationDate protobuf int64 field and then converting that back to time.Time on the target side.

I'm afraid that is a protocol break as all other places we use the Creation protobuf int64 field it represents seconds since 1970.

Reply:
Yup, this is how to get the nanoseconds:

```go
	now := time.Now()
	fmt.Println(now) // 2024-02-15 10:14:48.444236012 +0100 CET m=+0.000062533
	nowFromNano := time.Unix(0, now.UnixNano()) // 2024-02-15 10:14:48.444236012 +0100 CET
	fmt.Println(nowFromNano)
```

Reply:
What I suggest we do in order to maintain some modicum of backwards compatibility is to send the CreationDate as whole seconds (so we are being consistent with what that field represents in the protocol).

Then on the receiving side, when evaluating whether to refresh the snapshots or not, take the local snapshot creation time and then take only the whole seconds part and use that to compare to the creation time coming from the protobuf field.

Reply:
> Yup, this is how to get the nanoseconds:

I'm afraid we cannot do that.